closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -319,14 +345,13 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
                 crypto_library = NULL;
                 return -1;
             }
-            ossl_ver = 0;
         }
     } else {
         openssl_version = (*OSSL_version)(0); /* get OPENSSL_VERSION */
         /* Ensure the OpenSSL version is "OpenSSL 1.1.x" or "OpenSSL 3.x.x". */
-        if ((0 != strncmp(openssl_version, OPENSSL_VERSION_1_1, strlen(OPENSSL_VERSION_1_1)))
-        && (0 != strncmp(openssl_version, OPENSSL_VERSION_3_X, strlen(OPENSSL_VERSION_3_X)))
-        ) {
+        ossl_ver = ver_to_l(openssl_version);
+        if(!(ossl_ver >= OPENSSL_VERSION_1_1_0 && ossl_ver < OPENSSL_VERSION_2_0_0)
+        && !(ossl_ver >= OPENSSL_VERSION_3_0_0 && ossl_ver < OPENSSL_VERSION_4_0_0)) {
```

Comment:
Missing parentheses and space after `if`.

Reply:
Edited.

Reply:
Redundant test `(ossl_ver < 0)` here too. Please put `) {` on a separate line.

Reply:
Edited and removed `(ossl_ver < 0)`.